app/models/post.rb

```diff
@@ -1,5 +1,14 @@
-class Post < ApplicationRecord
-  belongs_to :user
-  has_many :likes
-  has_many :comments
+class Post < ActiveRecord::Base
+  belongs_to :user, class_name: 'User'
+  has_many :comments, class_name: 'Comment', foreign_key: 'post_id'
+  has_many :likes, class_name: 'Like', foreign_key: 'post_id'
+
+  def posts_counter
```

Comment:
Great job @j-tee 🤝

- [ ] [Optional] I would suggest you rename this `posts_counter` method to `update_post_counter`.
- [ ] [Optional] Kindly use one of the [Rails Active Callbacks](https://guides.rubyonrails.org/active_record_callbacks.html) - `after_save` to call the `#update_post_counter` method every time a new post has been saved to the database, then make the `update_post_counter` method private.
![post-counter](https://user-images.githubusercontent.com/63917725/213447021-b323132c-e0f1-44c8-8ce2-5b1b4f32995c.PNG).

- [ ] [Optional] Kindly repeat the above for both the `update_comment_counter` and `update_like_conter`.